packages/common/lib/types/StateManager.types.ts

```diff
@@ -4,6 +4,11 @@ export interface CredentialOfferState {
   clientId?: string;
   credentialOffer: CredentialOfferPayloadV1_0_11;
   createdOn: number;
+  'pre-authorized_code': string;
+  preAuthorizedCodeCreatedOn: number;
+  preAuthorizedCodeExpiresIn: number;
```

Comment:
I rather have a configurable expiration duration, which we use together with the createdOn to determine whether a token is expired. Means we do not need to store this property at all.

Reply:
This is actually duration, I'm just using the suffix expires in for the sake of consistency. BTW, I think we can read it as 'pre-authorized_code expires in 1 minute' or something. I'm not sure if it is semantically wrong.

Reply:
There is a validation rule:

invalid_grant:
      the Authorization Server expects a PIN in the pre-authorized flow but the client provides the wrong PIN
      the End-User provides the wrong Pre-Authorized Code or the **Pre-Authorized Code has expired**

Since the token expiration time not present in the [request](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-token-request), How should I check whether it is expired?

Reply:
Fixed!